### ISSUE TITLE
about the Matchability Map

### ISSUE THREAD

**Issue by xubin1994** [jzhangbs/DSM#1]
I read your article and used your training code to reproduce the results of the paper, but I feel that there seems to be a problem with the calculation of the Matchability Map. I analyzed the source code, and it seems that the uncetain loss of the code is a bit inconsistent with the original paper.
![image](https://user-images.githubusercontent.com/28856641/139365011-ada77e06-6b82-47e1-8482-80efc441d70d.png)
![image](https://user-images.githubusercontent.com/28856641/139365182-23ccd8b2-63ff-4f0d-9cc7-34dc6f9625b2.png)
Can you please explain why the Matchability map I got doesn't look right?

**Comment by xubin1994** [comment]
I re-looked at the code and the formula in the paper are consistent, but I still don’t know why my results look wrong

**Comment by jzhangbs** [comment]
Hi, if the images are plotted by the same piece of code, I think the problem is from training. How about the quantitative results of your trained model?

**Comment by xubin1994** [comment]
author param:I average, val_loss 0.8007 (0.9488 7.6520 9.4015), (<1px) 0.9198, (<3px) 0.9586, (D1) 3.4997
my param:I average, val_loss 1.1343 (1.2917 6.4861 8.9121), (<1px) 0.8912, (<3px) 0.9480, (D1) 4.5420
but looking at the disparity map is basically correct, except for the Matchability map, which is strange. Maybe it is because Matchability map is more difficult to converge?
In addition, is there a minus sign missing from the formula derivation in the paper?
![image](https://user-images.githubusercontent.com/28856641/139398997-55da0ed3-6e2c-435d-bf67-62a97abf0c63.png)
Should this be the case as below?
![image](https://user-images.githubusercontent.com/28856641/139398624-a01d7722-59a0-4ce0-913e-5600a53be337.png)

**Comment by jzhangbs** [comment]
There should be an additional minus in the exp of the first line. The second line is correct.

**Comment by xubin1994** [comment]
I don't have enough  memory to train batch = 16, so can training with batch = 2 for a long enough time also generate the correct Matchability Map?I fould that you wrote a refine_loss in the code, do you need to add it to training?
![image](https://user-images.githubusercontent.com/28856641/139408689-ee8f2fdf-8aba-42b4-bd91-7a4d16f62d10.png)
